modules/analysis/src/service.rs

```diff
@@ -273,6 +274,23 @@ pub async fn load_graphs(
     }
 }
 
+fn convert_query_to_hashmap(query: &Query) -> HashMap<String, String> {
+    let mut query_map = HashMap::new();
+    if query.q.contains('=') {
+        for pair in query.q.split('&') {
+            if let Some((key, mut value)) = pair.split_once('=') {
+                if value.starts_with("urn:uuid:") {
+                    value = value.strip_prefix("urn:uuid:").unwrap_or(value);
+                }
+                query_map.insert(key.to_owned(), value.to_owned());
+            }
+        }
+    } else {
+        query_map.insert("re_name".to_owned(), query.q.clone());
+    }
+    query_map
+}
```

Comment:
Now I understand why you were asking how to get the parsed value out of `Query`. If we find ourselves wanting this in other places, we can refactor to hang this fn off `Query`.

Minor style nit: I don't love the "mutable hashmap plus for loop" pattern here. Smells like old java. :stuck_out_tongue_winking_eye:  Next time, consider calling `reduce` or `fold` on the `split`.

Reply:
ah ya, I made this loop explicit as I could not figure out why things were not working -  then grokked the need to strip urn:uuid: for the graph query bit ... will rewrite, thx!!